cloudflare/resource_cloudflare_access_identity_provider.go

```diff
@@ -0,0 +1,329 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+const CONCEALED_STRING = "**********************************"
+
+func resourceCloudflareAccessIdentityProvider() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareAccessIdentityProviderCreate,
+		Read:   resourceCloudflareAccessIdentityProviderRead,
+		Update: resourceCloudflareAccessIdentityProviderUpdate,
+		Delete: resourceCloudflareAccessIdentityProviderDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareAccessIdentityProviderImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"centrify", "facebook", "google-apps", "oidc", "github", "google", "saml", "linkedin", "azureAD", "okta", "onetimepin", "onelogin", "yandex"}, false),
+			},
+			"config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"apps_domain": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"attributes": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"auth_url": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"centrify_account": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"centrify_app_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"certs_url": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"client_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"client_secret": {
+							Type:     schema.TypeString,
+							Optional: true,
+							// client_secret is a write only operation from the Cloudflare API
```

Comment:
I'm still not totally sold on this however it's the most elegant way of managing the write-but-never-retrievable situation the `client_secret` and `idp_public_cert` fields are in.